pulsar-broker/src/main/java/com/yahoo/pulsar/broker/loadbalance/impl/BrokerVersionFilter.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2016 Yahoo Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yahoo.pulsar.broker.loadbalance.impl;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.yahoo.pulsar.broker.BrokerData;
+import com.yahoo.pulsar.broker.loadbalance.BrokerFilter;
+import com.yahoo.pulsar.broker.BundleData;
+import com.yahoo.pulsar.broker.ServiceConfiguration;
+import com.yahoo.pulsar.broker.loadbalance.LoadData;
+
+public class BrokerVersionFilter implements BrokerFilter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GenericBrokerHostUsageImpl.class);
+
+    /**
+     * From the given set of available broker candidates, filter those using the load data.
+     *
+     * @param brokers
+     *            The currently available brokers that have not already been filtered. This set may be modified by
+     *            filter.
+     * @param bundleToAssign
+     *            The data for the bundle to assign.
+     * @param loadData
+     *            The load data from the leader broker.
+     * @param conf
+     *            The service configuration.
+     */
+    public void filter(Set<String> brokers, BundleData bundleToAssign, LoadData loadData, ServiceConfiguration conf) {
+        if ( conf.getActiveVersion() == null || conf.getActiveVersion().length() == 0 ) {
+            return;
+        }
+
+        final String activeVersion = conf.getActiveVersion();
+
+        Map<String, BrokerData> brokerData = loadData.getBrokerData();
+        List<String> brokersToRemove = new LinkedList<>();
+        for ( String broker : brokers ) {
```

Comment:
The reason why I didn't do this is because it is possible that the filter pipeline removes all candidates. E.g., when you set the active version to the next version just before a deployment and a new request comes in. 

However, I can call `LoadManagerShared.applyPolicies()` again if that is the case, so the copy will become the _exceptional_ case rather than the _normal_ case.